File: src/main/java/com/zps/zest/autocomplete/context/CompletionContext.java
package com.zps.zest.autocomplete.context;

import com.intellij.psi.*;
import java.util.List;

/**
 * Complete context information for code completion.
 */
public class CompletionContext {
    public final CursorPosition cursorPosition;
    public final ContainingElements containingElements;
    public final SemanticInfo semanticInfo;
    public final CompletionType completionType;
    public final LocalContext localContext;
    public final String languageLevel;
    
    public CompletionContext(CursorPosition cursorPosition, ContainingElements containingElements,
                           SemanticInfo semanticInfo, CompletionType completionType,
                           LocalContext localContext, String languageLevel) {
        this.cursorPosition = cursorPosition;
        this.containingElements = containingElements;
        this.semanticInfo = semanticInfo;
        this.completionType = completionType;
        this.localContext = localContext;
        this.languageLevel = languageLevel;
    }
}

File: src/main/kotlin/com/zps/zest/completion/data/CompletionContext.kt
package com.zps.zest.completion.data

import com.intellij.openapi.editor.Document
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.util.TextRange
import com.intellij.openapi.fileEditor.FileDocumentManager

/**
 * Simple context information for completion requests
 */
data class CompletionContext(
    val fileName: String,
    val language: String,
    val prefixCode: String,
    val suffixCode: String,
    val offset: Int,
    val manually: Boolean,
    val requestTime: Long = System.currentTimeMillis()
) {
    companion object {
        fun from(editor: Editor, offset: Int, manually: Boolean = false): CompletionContext {
            val document = editor.document
            val virtualFile = FileDocumentManager.getInstance().getFile(document)
            val fileName = virtualFile?.name ?: "unknown"
            val language = virtualFile?.fileType?.name ?: "text"
            
            val text = document.text
            
            // Simple prefix/suffix extraction
            val prefixStart = maxOf(0, offset - MAX_PREFIX_LENGTH)
            val suffixEnd = minOf(text.length, offset + MAX_SUFFIX_LENGTH)
            
            val prefixCode = text.substring(prefixStart, offset)
            val suffixCode = text.substring(offset, suffixEnd)
            
            return CompletionContext(
                fileName = fileName,
                language = language,
                prefixCode = prefixCode,
                suffixCode = suffixCode,
                offset = offset,
                manually = manually
            )
        }
        
        private const val MAX_PREFIX_LENGTH = 2000
        private const val MAX_SUFFIX_LENGTH = 500
    }
    
    /**
     * Get the line content at the cursor position
     */
    fun getCurrentLine(document: Document): String {
        val lineNumber = document.getLineNumber(offset)
        val lineStart = document.getLineStartOffset(lineNumber)
        val lineEnd = document.getLineEndOffset(lineNumber)
        return document.getText(TextRange(lineStart, lineEnd))
    }
    
    /**
     * Get the text before cursor on the current line
     */
    fun getLinePrefixText(document: Document): String {
        val lineNumber = document.getLineNumber(offset)
        val lineStart = document.getLineStartOffset(lineNumber)
        return document.getText(TextRange(lineStart, offset))
    }
    
    /**
     * Check if the cursor is at the beginning of a line
     */
    fun isAtLineStart(document: Document): Boolean {
        return getLinePrefixText(document).isBlank()
    }
}
